starcheck/src/starcheck.pl

```diff
@@ -790,41 +791,45 @@ sub json_obsids{
     $out .= sprintf "<A HREF=\"#obsid$obs{$obsid}->{obsid}\">OBSID = %5s</A>", $obs{$obsid}->{obsid};
     $out .= sprintf " at $obs{$obsid}->{date}   ";
 
-    my $good_guide_count = $obs{$obsid}->{count_nowarn_stars}{GUI};
-    my $good_acq_count = $obs{$obsid}->{count_nowarn_stars}{ACQ};
+    my $guide_count = $obs{$obsid}->count_guide_stars();
 
     # if Obsid is numeric, print tally info
     if ($obs{$obsid}->{obsid} =~ /^\d+$/ ){
 
-        # minumum requirements for acq and guide for ERs and ORs
-        # should be set by config...
-        my $min_num_acq = ($obs{$obsid}->{obsid} >= 38000 ) ? 5 : 4;
-        my $min_num_gui = ($obs{$obsid}->{obsid} >= 38000 ) ? 6 : 4;
+        # minumum requirements for fractional guide star count for ERs and ORs
+        my $min_num_gui = ($obs{$obsid}->{obsid} >= 38000 ) ? 6.0 : 4.0;
 
         # if there is no star catalog and that's ok
         if (not ($obs{$obsid}->find_command("MP_STARCAT"))
-            and $obs{$obsid}->{ok_no_starcat}){
-            $min_num_acq = 0;
-            $min_num_gui = 0;
+                and $obs{$obsid}->{ok_no_starcat}){
+            $min_num_gui = 0.0;
         }
 
         # use the 'special case' ER rules from ACA-044
         if ($obs{$obsid}->{special_case_er}){
-            $min_num_acq = 4;
-            $min_num_gui = 4;
+            $min_num_gui = 4.0;
         }
 
-        my $acq_font_start = ($good_acq_count < $min_num_acq) ? $red_font_start
+        # Use the acq prob model values saved in figure_of_merit for the expected
+        # number of acq stars and a bad overall probability.  figure_of_merit isn't
+        # defined if there is no star catalog, so use default of 0 stars and not-bad (0 status)
+        my $n_acq = 0.0;
+        my $bad_acq_prob = 0;
+        if (defined $obs{$obsid}->{figure_of_merit}){
+            $n_acq = $obs{$obsid}->{figure_of_merit}->{expected};
+            $bad_acq_prob = $obs{$obsid}->{figure_of_merit}->{cum_prob_bad};
+        }
+        my $acq_font_start =  $bad_acq_prob ? $red_font_start
         : $empty_font_start;
-        my $gui_font_start = ($good_guide_count < $min_num_gui) ? $red_font_start
+        my $gui_font_start = ($guide_count < $min_num_gui) ? $red_font_start
         : $empty_font_start;
 
         $out .= "$acq_font_start";
-        $out .= sprintf "$good_acq_count clean ACQ | ";
+        $out .= sprintf("%3.1f ACQ | ", $n_acq);
```

Comment:
If there is no catalog would it make sense to change the output to `---` (with appropriate padding spaces for formatting) instead of `0.0 ACQ` or `0.0 GUI`.

Reply:
It might help to see this run for a week with a dark cal.  I guess there will still be two red warnings but with obvious disposition.

Reply:
For dark cals there is the dream that they will eventually be in the "ok_no_starcat" block again.  IIRC, when ORViewer got promoted and I no longer had very clear dark current OFLS ids, fewer observations got correctly tagged as "ok_no_starcat".  But sure, I can put up the current dark cal output.

Reply:
Regarding "If there is no catalog would it make sense to change the output to ---"  I'm not sure.  Did you want to reorganize this routine to handle the printing of the lines for observations without star catalogs in a new block?

Reply:
I think I just fixed the ok_no_starcat in a way that makes sense (used the list of observations from the dark cal checker that are known to actually *be* the dark cal observations as a new set that should be marked "ok_no_starcat").